src/Azure.Functions.Cli/Helpers/PythonHelpers.cs

```diff
@@ -73,85 +76,100 @@ private static void CreateRequirements()
+            // If any of the possible python executables are 3.x, we will take that.
+            WorkerLanguageVersionInfo python3worker = versions.FirstOrDefault(w => w?.Major == 3);
             if (python3worker != null)
             {
-                if (errorIfNoExactMatch) throw new CliException(string.Format(errorIfNotExactMessage, python3worker.Version));
-                SetWorkerPathIfNeeded(setWorkerExecutable, python3worker.ExecutablePath);
-                ColoredConsole.WriteLine(WarningColor(string.Format(warningMessage, python3worker.Version)));
-                return python3worker;
-            }
-
-            // If we found any python versions at all, we warn or error out if flag enabled.
-            WorkerLanguageVersionInfo anyPythonWorker = workers.FirstOrDefault(w => !string.IsNullOrEmpty(w?.Version));
-            if (anyPythonWorker != null)
-            {
-                if (errorIfNoExactMatch) throw new CliException(string.Format(errorIfNotExactMessage, anyPythonWorker.Version));
-                if (errorOutIfOld) throw new CliException(string.Format(errorMessageOldPy, string.Join(", ", anyPythonWorker.Version)));
-                else ColoredConsole.WriteLine(WarningColor(string.Format(errorMessageOldPy, string.Join(", ", anyPythonWorker.Version))));
+                _pythonVersionCache = python3worker;
+                return _pythonVersionCache;
             }
 
-            // If we didn't find python at all, we warn or error out if flag enabled.
-            else
-            {
-                if (errorOutIfOld) throw new CliException(errorMessageNoPy);
-                else ColoredConsole.WriteLine(WarningColor(errorMessageNoPy));
-            }
+            // Least preferred -- If we found any python versions at all we return that
+            WorkerLanguageVersionInfo anyPythonWorker = versions.FirstOrDefault(w => !string.IsNullOrEmpty(w?.Version));
+            _pythonVersionCache = anyPythonWorker ?? new WorkerLanguageVersionInfo(WorkerRuntime.python, null, null);
 
-            return null;
+            return _pythonVersionCache;
         }
 
-        private static void SetWorkerPathIfNeeded(bool setWorker, string pyExe)
+        public static void SetWorkerPath(string pyExe, bool overwrite = false)
         {
-            if (setWorker)
+            if (overwrite || string.IsNullOrEmpty(Environment.GetEnvironmentVariable(_pythonDefaultExecutableVar)))
             {
                 Environment.SetEnvironmentVariable(_pythonDefaultExecutableVar, pyExe, EnvironmentVariableTarget.Process);
-                if (StaticSettings.IsDebug)
-                {
-                    ColoredConsole.WriteLine(VerboseColor($"{_pythonDefaultExecutableVar} set to {pyExe}"));
-                }
+            }
+            if (StaticSettings.IsDebug)
+            {
+                ColoredConsole.WriteLine(VerboseColor($"{_pythonDefaultExecutableVar} set to {pyExe}"));
```

Comment:
I think when overwrite is set to False, and _pythonDefaultExecutableVar has value, the debug log will not reflects the actual _pythonDefaultExecutableVar value since pyExe is not set to the worker path.

Reply:
Right, I will change the wording to be `$"{_pythonDefaultExecutableVar} is set to {pyExe}"`
Still good to show in debug what it's actually set to.

Reply:
Then I think the second value needs to change to **pyExe** -> **Environment.GetEnvironmentVariable(_pythonDefaultExecutableVar)**.

Reply:
Right right. duh!